src/Library/demos/Event Controllers/main.js

```diff
@@ -0,0 +1,103 @@
+const stack = workbench.builder.get_object("stack");
+const pic1 = workbench.builder.get_object("pic1");
+const pic2 = workbench.builder.get_object("pic2");
+const gesture_frame = workbench.builder.get_object("gesture_frame");
+const primary_button = workbench.builder.get_object("primary_button");
+const middle_button = workbench.builder.get_object("middle_button");
+const secondary_button = workbench.builder.get_object("secondary_button");
+
+const pic1_file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos/Event Controllers/image1.png",
+);
+
+const pic2_file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos/Event Controllers/image2.png",
+);
+
+pic1.file = pic1_file;
+pic2.file = pic2_file;
+
+let ctrl_pressed = false;
+
+// Key controller to detect when the Ctrl key is pressed and released
+const key_controller = new Gtk.EventControllerKey();
+window.add_controller(key_controller);
+key_controller.connect("key-pressed", (controller, keyval, keycode, state) => {
+  if (keyval === Gdk.KEY_Control_L || keyval === Gdk.KEY_Control_R) {
+    ctrl_pressed = true;
+  }
+});
+
+key_controller.connect("key-released", (controller, keyval, keycode, state) => {
+  if (keyval === Gdk.KEY_Control_L || keyval === Gdk.KEY_Control_R) {
+    ctrl_pressed = false;
+  }
+});
+
+// Gesture controller to detect when the button is clicked
+const gesture_primary_click = new Gtk.GestureClick();
+
+ctrl_button.connect("clicked", () => {
+  if (ctrl_pressed) {
+    ctrl_button.label = _("Ctrl+Click detected");
+    ctrl_button.add_css_class("suggested-action");
+  } else {
+    ctrl_button.label = _("Click detected");
+    ctrl_button.remove_css_class("suggested-action");
+  }
+});
+
+const gesture_click = new Gtk.GestureClick({ button: 0 });
+
+window.add_controller(gesture_click);
+
+gesture_click.connect("pressed", (gesture, n_press, x, y) => {
+  switch (gesture.get_current_button()) {
+    case Gdk.BUTTON_PRIMARY:
+      primary_button.add_css_class("suggested-action");
+      break;
+
+    case Gdk.BUTTON_MIDDLE:
+      middle_button.add_css_class("suggested-action");
+      break;
+
+    case Gdk.BUTTON_SECONDARY:
+      secondary_button.add_css_class("suggested-action");
+      break;
+  }
+});
+
+gesture_click.connect("released", (gesture, n_press, x, y) => {
+  switch (gesture.get_current_button()) {
+    case Gdk.BUTTON_PRIMARY:
+      primary_button.remove_css_class("suggested-action");
+      break;
+
+    case Gdk.BUTTON_MIDDLE:
+      middle_button.remove_css_class("suggested-action");
+      break;
+
+    case Gdk.BUTTON_SECONDARY:
+      secondary_button.remove_css_class("suggested-action");
+      break;
+  }
+});
+
+const single = new Gtk.GestureSwipe();
+
+stack.add_controller(single);
+
+single.connect("swipe", (controller, vel_x, vel_y) => {
+  if (vel_x > 0) {
+    stack.set_visible_child_name("pic1");
+  } else {
+    stack.set_visible_child_name("pic2");
+  }
+});
```

Comment:
This is a great, simple demonstration of this event controller. The only problem I have is that there's no indication which way you can swipe. Maybe add a simple `Gtk.StackSwitcher` above/below with "Start" and "Finish" labels for the pages?